File: app/src/main/java/tech/cherri/tokenpushexample/TokenPushParam.java
package tech.cherri.tokenpushexample;


import android.content.Context;

import tech.cherri.tokenpushexample.callback.TokenPushCallback;

public class TokenPushParam {
    private Context context;
    private TokenPushCallback tokenPushCallback;
    private String tspToken;
    private String cancelUrl;

    public TokenPushParam(TokenPushParamBuilder tokenPushParamBuilder) {
        tokenPushCallback = tokenPushParamBuilder.tokenPushCallback;
        context = tokenPushParamBuilder.context;
        tspToken = tokenPushParamBuilder.tspToken;
        cancelUrl = tokenPushParamBuilder.cancelUrl;
    }

    public static class TokenPushParamBuilder {
        private Context context;
        private TokenPushCallback tokenPushCallback;
        private String tspToken;
        private String cancelUrl;

        public TokenPushParamBuilder context(Context context) {
            this.context = context;
            return this;
        }

        public TokenPushParamBuilder tokenPushCallback(TokenPushCallback tokenPushCallback) {
            this.tokenPushCallback = tokenPushCallback;
            return this;
        }

        public TokenPushParamBuilder tspToken(String tspToken) {
            this.tspToken = tspToken;
            return this;
        }

        public TokenPushParamBuilder cancelUrl(String cancelUrl) {
            this.cancelUrl = cancelUrl;
            return this;
        }

        public TokenPushParam build() {
            return new TokenPushParam(this);
        }
    }


    public Context getContext() {
        return context;
    }

    public TokenPushCallback getTokenPushCallback() {
        return tokenPushCallback;
    }

    public String getTspToken() {
        return tspToken;
    }

    public String getCancelUrl() {
        return cancelUrl;
    }

}

File: kotlin/app/src/main/java/tech/cherri/tokenpushexample/TokenPushParam.kt
package tech.cherri.tokenpushexample

import android.content.Context
import tech.cherri.tokenpushexample.callback.TokenPushCallback

class TokenPushParam(tokenPushParamBuilder: TokenPushParamBuilder) {

    var context: Context?
    var tokenPushCallback: TokenPushCallback?
    var tspToken: String?
    var cancelUrl: String?

    class TokenPushParamBuilder {
        var context: Context? = null
        var tokenPushCallback: TokenPushCallback? = null
        var tspToken: String? = null
        var cancelUrl: String? = null
        fun context(context: Context?): TokenPushParamBuilder {
            this.context = context
            return this
        }

        fun tokenPushCallback(tokenPushCallback: TokenPushCallback?): TokenPushParamBuilder {
            this.tokenPushCallback = tokenPushCallback
            return this
        }

        fun tspToken(tspToken: String?): TokenPushParamBuilder {
            this.tspToken = tspToken
            return this
        }

        fun cancelUrl(cancelUrl: String?): TokenPushParamBuilder {
            this.cancelUrl = cancelUrl
            return this
        }

        fun build(): TokenPushParam {
            return TokenPushParam(this)
        }
    }

    init {
        tokenPushCallback = tokenPushParamBuilder.tokenPushCallback
        context = tokenPushParamBuilder.context
        tspToken = tokenPushParamBuilder.tspToken
        cancelUrl = tokenPushParamBuilder.cancelUrl
    }

}